packages/vue-renderer/src/renderer.js

```diff
@@ -499,7 +282,7 @@ export default class VueRenderer {
 
     // context.modern
     if (context.modern === undefined) {
-      context.modern = req.modernMode && this.context.options.modern === 'server'
+      context.modern = req._modern || this.context.options.modern === 'client'
```

Comment:
Could be `modern` indeed, my consideration is avoiding potential override if any 3rd party lib/plugin injects `modern` on `req` since `modern` is a common word

Reply:
for nuxt3 we can probably prefix all custom configs in `req` with `nuxt`. OK. Let's keep it as is for now :)